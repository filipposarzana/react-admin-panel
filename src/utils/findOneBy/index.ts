import { User, users } from '~/db'

type Key = Extract<keyof User, 'name' | 'uuid'>

const findOneBy = <K extends Key>(prop: K) => (value: User[K]) =>
  users.find((user) => user[prop].toLowerCase() === value.toLowerCase())

export const findOneByName = findOneBy('name')

export const findOneByUuid = findOneBy('uuid')
