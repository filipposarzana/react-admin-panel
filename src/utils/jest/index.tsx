import { render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { ReactElement } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { StyleSheetManager } from 'styled-components'

export { act, fireEvent, waitFor } from '@testing-library/react'

export const renderer = {
  deep: (node: ReactElement) => {
    const r = render(<StyleSheetManager disableVendorPrefixes>{node}</StyleSheetManager>)

    return {
      ...r,
      output: r.container.firstChild,
    }
  },
  hook: <T extends unknown, R extends unknown>(hookFunction: (props: T) => R) => {
    const r = renderHook(hookFunction)

    return r
  },
  shallow: (node: ReactElement) => {
    const r = createRenderer()

    r.render(node)

    return {
      instance: r.getMountedInstance(),
      output: r.getRenderOutput(),
      unmount: r.unmount,
    }
  },
}
