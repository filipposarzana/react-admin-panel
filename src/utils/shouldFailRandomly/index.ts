export const shouldFailRandomly = () => Math.round(Math.random()) === 0
