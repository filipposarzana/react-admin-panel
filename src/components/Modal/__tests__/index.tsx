import { renderer } from '~/utils/jest'
import { Modal, Props } from '..'

const fixtures: [string, Props][] = [
  ['matches snapshot when closed', { index: 0, onClose: jest.fn(), open: false }],
  ['matches snapshot with open', { index: 0, onClose: jest.fn(), open: true }],
]

jest.mock('~/components/Page/Create')

afterAll(jest.clearAllMocks)

describe('Modal', () => {
  fixtures.forEach(([title, props]) => {
    it(title, () => {
      const { output } = renderer.shallow(<Modal {...props} />)

      expect(output).toMatchSnapshot()
    })
  })

  it('matches style snapshot', () => {
    const { output } = renderer.deep(<Modal index={0} onClose={jest.fn()} open />)

    expect(output).toMatchSnapshot()
  })
})
