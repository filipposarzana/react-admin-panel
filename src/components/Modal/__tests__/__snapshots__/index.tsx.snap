// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot when closed 1`] = `null`;

exports[`Modal matches snapshot with open 1`] = `
<React.Fragment>
  <Styled(styled.div)
    as="section"
    index={0}
    onClick={[Function]}
  />
  <Styled(styled.div)
    background="white"
    grow={1}
    index={0}
    p={16}
    shrink={1}
  >
    <styled.div
      align="flex-start"
      pb={16}
    >
      <Button
        onClick={[Function]}
        type="button"
      >
        Close
      </Button>
    </styled.div>
    <CreatePage
      index={0}
      onCreated={[MockFunction]}
    />
  </Styled(styled.div)>
</React.Fragment>
`;

exports[`Modal matches style snapshot 1`] = `
.c0 {
  display: flex;
  min-height: 0;
  min-width: 0;
  align-items: stretch;
  flex-basis: auto;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-start;
  flex-shrink: 0;
  flex-wrap: nowrap;
}

.c1 {
  background-color: rgba(0,0,0,0.4);
  height: 100%;
  left: 0px;
  position: fixed;
  top: 0px;
  width: 100%;
  z-index: 100;
}

<section
  class="c0 c1"
  direction="column"
  wrap="nowrap"
/>
`;
